File: java/playwright/src/test/java/dev/selenium/request/MockResponseTest.java
package dev.selenium.request;

import com.microsoft.playwright.Route;
import dev.selenium.TestBase;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class MockResponseTest extends TestBase {

  @Test
  public void mockResponseBody() {
    String mockBody = "<h1>Mock Response</h1>";
    page.route(
        "https://selenium.dev/",
        route -> {
          route.fulfill(new Route.FulfillOptions().setBody(mockBody));
        });

    page.navigate("https://selenium.dev");
    Assertions.assertEquals("Mock Response", page.textContent("h1"));
  }
}

File: dotnet/Playwright/Request/MockResponseTest.cs
using Microsoft.Playwright;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace Playwright.Request;

[TestClass]
public class MockResponseTest : TestBase
{
    [TestMethod]
    public async Task MockResponseBody()
    {
        var mockBody = "<h1>Mock Response</h1>";
        await Page!.RouteAsync("https://selenium.dev/", async route =>
        {
            await route.FulfillAsync(new RouteFulfillOptions { Body = mockBody });
        });

        await Page.GotoAsync("https://selenium.dev");
        var h1Text = await Page.TextContentAsync("h1");
        Assert.AreEqual("Mock Response", h1Text);
    }
}